src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -146,23 +221,43 @@ private static boolean isBoundary(final AtlasEntity crossingLine)
     public LineCrossingWaterBodyCheck(final Configuration configuration)
     {
         super(configuration);
+        this.filterOffendingLineItems = this.configurationValue(configuration,
+                "intersectingLineItems.filter", DEFAULT_OFFENDING_LINEITEM_FILTER);
     }
 
     @Override
     public boolean validCheckForObject(final AtlasObject object)
     {
-        return TypePredicates.IS_AREA.test(object) && TagPredicates.IS_WATER_BODY.test(object);
+        // We only consider water body areas, not linear water bodies
+        return TypePredicates.IS_AREA.test(object) && !INVALID_WATER_BODY_TAGS.test(object)
+                && VALID_WATER_BODY_TAGS.test(object);
     }
 
     @Override
     protected Optional<CheckFlag> flag(final AtlasObject object)
     {
-        // First retrieve the crossing edges and lines
+        // First retrieve the crossing edges, lines, buildings
         final Area objectAsArea = (Area) object;
         final Polygon areaAsPolygon = objectAsArea.asPolygon();
         final Atlas atlas = object.getAtlas();
-        final Iterable<AtlasItem> crossingLinearItems = new MultiIterable<>(
-                atlas.edgesIntersecting(areaAsPolygon), atlas.linesIntersecting(areaAsPolygon));
+        final Iterable<AtlasItem> allCrossingItems = this.filterOffendingLineItems
+                // If we're only looking for crossing buildings/railways/streets
+                ? new MultiIterable<>(
+                        atlas.lineItemsIntersecting(areaAsPolygon,
+                                lineItem -> (lineItem instanceof Edge
+                                        && HIGHWAY_TYPE_TO_FLAG.test((Edge) lineItem)
```

Comment:
nit: If we can move all of these validations into a method and call it from here will have increase readability and easy to test.

Reply:
Yup, good thinking